recipes_source/recipes/timer_quick_start.py

```diff
@@ -0,0 +1,461 @@
+        x = torch.ones((128,))
+        y = torch.ones(({b_size},))
+        """,
+    ).blocked_autorange(min_run_time=1)
+
+print(measure(b_size=128), "\n")
+print(measure(b_size=1), "\n")
+
+###############################################################################
+# .. code-block:: none
+#    :caption: Output
+#
+#     (The top half of the __repr__ has been trimmed for convenience.)
+#
+#     Case A: {128} x {128}
+#       Median: 2.59 us
+#       IQR:    0.09 us (2.55 to 2.64)
+#       381 measurements, 1000 runs per measurement, 1 thread
+#
+#     Case B: {128} x {1}
+#       Median: 2.69 us
+#       IQR:    0.08 us (2.65 to 2.74)
+#       366 measurements, 1000 runs per measurement, 1 thread
+#
+
+
+###############################################################################
+# From the look of it, the second (simpler) case is slower but it's hard to
+# tell if it's real or within the noise. For lower variance measurements we
+# can rerun the benchmark using the C++ frontend:
+#
+
+def measure_cpp(b_size):
+    return Timer(
+        "x * y;",
+        f"""
+        auto x = torch::ones({{128}});
+        auto y = torch::ones({{{b_size}}});
+        """,
+        language=Language.CPP,
+    ).blocked_autorange(min_run_time=1)
+
+print(measure_cpp(b_size=128), "\n")
+print(measure_cpp(b_size=1), "\n")
+
+###############################################################################
+# .. code-block:: none
+#    :caption: Output
+#
+#     Case A: {128} x {128}
+#       Median: 1.47 us
+#       IQR:    0.02 us (1.46 to 1.48)
+#       68 measurements, 10000 runs per measurement, 1 thread
+#
+#     Case B: {128} x {1}
+#       Median: 1.60 us
+#       IQR:    0.04 us (1.58 to 1.62)
+#       60 measurements, 10000 runs per measurement, 1 thread
+#
+
+###############################################################################
+# The C++ wall times confirm that what we're seeing isn't just noise, but they
+# do nothing to illuminate what is happening. For that, let's count
+# instructions using Callgrind.
+
+def measure_cpp_callgrind(b_size):
+    return Timer(
+        "x * y;",
+        f"""
+        auto x = torch::ones({{128}});
+        auto y = torch::ones({{{b_size}}});
+        """,
+        language=Language.CPP,
+    ).collect_callgrind()
+
+for _ in range(3):
+    print(measure_cpp_callgrind(128).counts())
+
+print()
+for _ in range(3):
+    print(measure_cpp_callgrind(1).counts())
+
+###############################################################################
+# .. code-block:: none
+#    :caption: Output
+#
+#     587972
+#     587972
+#     587972
+#
+#     646262
+#     646262
+#     646262
+#
+
+###############################################################################
+# As you can see, one of the selling points of Callgrind is that it is 100%
+# deterministic. This is very useful because it lets us A/B test very subtle
+# changes. Let's take a look at the CallgrindStats object that Timer returns:
```

Comment:
Does it make sense to mention that you need REL_WITH_DEB_INFO for detailed counts info? A warning will tell you if you don't have it, but still, having it here also makes sense to me.

Reply:
I've added a footnote section and included it.